File: crates/yamind-commands/src/command.rs
use yamind_core::Document;

/// A reversible command that can be executed and undone.
pub trait Command: std::fmt::Debug {
    fn execute(&mut self, doc: &mut Document);
    fn undo(&mut self, doc: &mut Document);
    fn description(&self) -> &str;
    /// Update the text associated with this command (for new-node edit flow).
    fn set_text(&mut self, _text: String) {}
}

File: src/src/shared/commands/command.ts
import type { Document } from '../types/document'

export interface Command {
  readonly type: string
  execute(doc: Document): void
  undo(doc: Document): void
}

/** Commands that support updateText (for new-node workflow) */
export interface TextUpdatable {
  updateText(newText: string): void
}

export function isTextUpdatable(cmd: Command): cmd is Command & TextUpdatable {
  return 'updateText' in cmd
}
